File: Documentation/3.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Session/HowTo/HasChanged.cs
using Raven.Client.Document;
using Raven.Documentation.CodeSamples.Orders;

namespace Raven.Documentation.Samples.ClientApi.Session.HowTo
{
	public class HasChanged
	{
		private interface IFoo
		{
			#region has_changed_1
			bool HasChanged(object entity);
			#endregion
		}

		public HasChanged()
		{
			using (var store = new DocumentStore())
			{
				using (var session = store.OpenSession())
				{
					#region has_changed_2
					var employee = session.Load<Employee>("employees/1");
					var hasChanged = session.Advanced.HasChanged(employee); // false
					employee.LastName = "Shmoe";
					hasChanged = session.Advanced.HasChanged(employee); // true
					#endregion
				}
			}
		}
	}
}

File: Documentation/5.4/Samples/nodejs/client-api/session/howTo/hasChanged.js
import { DocumentStore } from "ravendb";

const documentStore = new DocumentStore();
const session = documentStore.openSession();

{
    let entity;
    //region has_changed_1
    session.advanced.hasChanged(entity);
    //endregion
}

async function example() {
    {
        //region has_changed_2
        const employee = await session.load("employees/1-A");
        let hasChanged = session.advanced.hasChanged(employee); // false
        employee.lastName = "Shmoe";
        hasChanged = session.advanced.hasChanged(employee);       // true
        //endregion
    }
}
